src/main/java/net/dv8tion/jda/api/JDA.java

```diff
@@ -492,6 +489,47 @@ default JDA awaitReady() throws InterruptedException
     @Nonnull
     List<Object> getRegisteredListeners();
 
+    // TODO: Docs
+    @Nonnull
+    @CheckReturnValue
+    RestAction<List<Command>> retrieveCommands();
+
+    @Nonnull
+    @CheckReturnValue
+    RestAction<Command> retrieveCommandById(@Nonnull String id);
+
+    @Nonnull
+    @CheckReturnValue
+    default RestAction<Command> retrieveCommandById(long id)
+    {
+        return retrieveCommandById(Long.toUnsignedString(id));
+    }
+
+    @Nonnull
+    @CheckReturnValue
+    CommandCreateAction upsertCommand(@Nonnull String name, @Nonnull String description);
```

Comment:
Maybe add `CommandCreateAction upsertCommand(@Nonnull CommandData command);` for convenience